src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -250,11 +267,18 @@ abstract class AbstractCheckTask(
      * @param checkpointTimestamp checkpoint timestamp from the previous task
      * @param executorService executor to schedule pipeline execution.
      * @param untrusted flag is guarantee that the previous sequence data is correct
+     * @param parentTaskCompleted indicates whether the parent task was completed normally. `null` if no parent task exists.
      * @throws IllegalStateException when method is called more than once.
      */
-    private fun begin(sequence: Long = DEFAULT_SEQUENCE, checkpointTimestamp: Timestamp? = null,
-                      executorService: ExecutorService = createExecutorService(), untrusted: Boolean = false) {
+    private fun begin(
+        sequence: Long = DEFAULT_SEQUENCE,
+        checkpointTimestamp: Timestamp? = null,
+        executorService: ExecutorService = createExecutorService(),
+        untrusted: Boolean = false,
+        parentTaskCompleted: Boolean? = null
```

Comment:
What about a new data class? Something like `ExecutionState`

Reply:
Do you mean replacing the `untrusted` and `parentTaskCompleted` parameters with that data class?